### PR TITLE
Correctly count methods when using declarative `#test` in minitest

### LINKED ISSUE #18408: Fix `rake stats` LOC/M counting when using testing DSL methods
If you use the following test form:

```ruby
test "something do
  assert
end
```

Then rake stats won't do proper LOC/M calculations. So you end up with something like:

```
| Controller tests     |  1350 |  1037 |      49 |       7 |   0 |   146 |
| Helper tests         |   295 |   221 |       9 |       3 |   0 |    71 |
| Model tests          |  1500 |  1151 |      58 |       9 |   0 |   125 |
| Mailer tests         |    22 |    20 |       3 |       4 |   1 |     3 |
| Job tests            |    48 |    39 |       3 |       0 |   0 |     0 |
| Integration tests    |   141 |   114 |       4 |       1 |   0 |   112 |
```

Let's fix that.

### PR DESCRIPTION
**Pull request by brainopia**
Fixes #18408
cc @dhh

### CODE REVIEW

**Comment by brainopia** [comment]
I've noticed there are a couple of old entries in the list of STATS_DIRECTORIES, they can be safely removed probably? 

```ruby
  %w(Functional\ tests\ (old)  test/functional),
  %w(Unit\ tests \ (old)       test/unit)
```

**Comment by dhh** [comment]
Yeah, we can kill those now! Otherwise looking good :+1:

File: railties/lib/rails/code_statistics_calculator.rb [hunk 1]
```diff
@@ -73,7 +75,9 @@ def add_by_io(io, file_type)
   end
 
   private
+
```
Comment (dhh) [inline railties/lib/rails/code_statistics_calculator.rb]: :scissors: – no newline after the private flag.

File: railties/lib/rails/code_statistics_calculator.rb [hunk 2]
```diff
@@ -73,7 +75,9 @@ def add_by_io(io, file_type)
...
     def file_type(file_path)
+      return :minitest if file_path.end_with? '_test.rb'
```
Comment (dhh) [inline railties/lib/rails/code_statistics_calculator.rb]: I'd prefer this as a full if/else, so:

```ruby
if file_path.end_with? '_test.rb'
  :minitest
else
  File.extname(file_path).sub(/\A\./, '').downcase.to_sym
end
```

File: railties/test/code_statistics_calculator_test.rb [hunk 3]
```diff
@@ -6,6 +6,55 @@ def setup
     @code_statistics_calculator = CodeStatisticsCalculator.new
   end
 
+  def temp_file(name, content)
```
Comment (dhh) [inline railties/test/code_statistics_calculator_test.rb]: Test helpers should go at the bottom as private methods.

**Comment by dhh** [comment]
Great work!

**Comment by carlosantoniodasilva** [comment]
:green_heart::yellow_heart::heart::blue_heart::purple_heart:

**Pull request merged**